src/abstractitensornetwork.jl

```diff
@@ -721,6 +721,48 @@ function contract_inner(
   return contract(tn; sequence)[]
 end
 
+function contract_with_BP(
+  itn::AbstractITensorNetwork; outputlevel=1, partitioning=group(v -> v, vertices(itn))
+)
+  return contract_with_BP(
+    ComplexF64, itn::AbstractITensorNetwork; outputlevel, partitioning
+  )
+end
+
+function contract_with_BP(
+  T::Type,
+  itn::AbstractITensorNetwork;
+  outputlevel=1,
+  partitioning=group(v -> v, vertices(itn)),
+)
+  @assert isempty(externalinds(itn))
+  bp_cache = BeliefPropagationCache(copy(itn), partitioning)
+
+  bp_cache = update(bp_cache)
+
+  pg = partitioned_itensornetwork(bp_cache)
+
+  if !is_tree(partitioned_graph(pg)) && outputlevel > 0
+    println("Partitioned graph is not a tree, result will be approximate!!")
+  end
```

Comment:
Let's remove this, I think this is inherent if someone selects that they are using BP.

Reply:
Yes, but it may be good to allow to pass a flag like `assert_exact=true`, to check that the `BP` will indeed be exact for the contraction (for the case of contracting `TTN`s)?

Reply:
Let's think about the design of that, but I don't think we have to address that in this PR. I agree with the sentiment that we don't want to secretly perform approximate contractions but give users the impression that we are doing exact contractions.